src/shared/cache-types.cts

```diff
@@ -44,17 +45,19 @@ type IncrementalCachedAppPageValueForMultipleVersions = Omit<
   'kind'
 > & {
   kind: 'APP_PAGE'
-}
+} & Pick<IncrementalCachedAppPageValueForNewVersions, 'segmentData'>
```

Comment:
This is terrible (stitching type from different node versions, but if I just tried to use types for newer version, all hell broke lose, so this, while awful, felt like only reasonable option without getting stuck in fixing bunch of types